### PR TITLE
refactor: restore JSCallInvoker

### PR DESCRIPTION
**Pull request by tjzel**
## Summary

Turns out that majority of work with regards to using `RuntimeExecutor` done in 
- https://github.com/software-mansion/react-native-reanimated/pull/5734
can be reverted since `jsCallInvoker` is staying in the API.

Thanks to that we can significantly cleanup the code which is helpful for Worklets.

## Test plan

I tested the changes on 0.74, 0.75 and 0.76 for Android/iOS Paper/Fabric and all worked well.

### CODE REVIEW

File: packages/react-native-reanimated/android/src/main/java/com/swmansion/worklets/JSCallInvokerResolver.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    try {
+      var catalystInstance = context.getClass().getMethod("getCatalystInstance").invoke(context);
+      assert catalystInstance != null;
+      var method = catalystInstance.getClass().getMethod("getJSCallInvokerHolder");
+      return (CallInvokerHolderImpl) method.invoke(catalystInstance);
+    } catch (Exception e) {
+      throw new RuntimeException("Failed to get JSCallInvokerHolder", e);
+    }
+  }
+}
```
Comment (tjzel) [inline packages/react-native-reanimated/android/src/main/java/com/swmansion/worklets/JSCallInvokerResolver.java:27]: In my opinion it's more maintainable to use a reflection in this case compared to source sets. I will add relevant comments here.

File: packages/react-native-reanimated/apple/reanimated/apple/REAModule.mm [hunk 2]
```diff
@@ -279,38 +276,18 @@ - (void)sendEventWithName:(NSString *)eventName body:(id)body
...
+  auto jsCallInvoker = self.bridge.jsCallInvoker;
+  auto jsiRuntime = reinterpret_cast<facebook::jsi::Runtime *>(self.bridge.runtime);
+
+  assert(jsiRuntime != nullptr);
+
+  auto reanimatedModuleProxy =
+      reanimated::createReanimatedModule(self, self.bridge, jsCallInvoker, workletsModule, _isBridgeless);
+
+  jsi::Runtime &rnRuntime = *jsiRuntime;
+  [self commonInit:reanimatedModuleProxy withRnRuntime:rnRuntime];
```
Comment (tomekzaw) [inline packages/react-native-reanimated/apple/reanimated/apple/REAModule.mm:289]: We shouldn't use `self.bridge.jsCallInvoker` in bridgeless mode. You have to do the following:
```objc
#ifdef RCT_NEW_ARCH_ENABLED
@synthesize callInvoker = _callInvoker;
#endif // RCT_NEW_ARCH_ENABLED

auto jsCallInvoker = _callInvoker.callInvoker;
```

Let's restore the original implementation without refactoring the whole function. See https://github.com/software-mansion/react-native-reanimated/pull/6768/files#diff-06f7028a5f8a14ac525b70098bbf48937abe91707aea813bc4528ec9c166f5bc

File: packages/react-native-reanimated/apple/reanimated/apple/REAModule.mm [hunk 3]
```diff
@@ -327,6 +304,7 @@ - (void)commonInit:(std::shared_ptr<ReanimatedModuleProxy>)reanimatedModuleProxy
   WorkletRuntimeCollector::install(rnRuntime);
   RNRuntimeDecorator::decorate(rnRuntime, reanimatedModuleProxy);
 #ifdef RCT_NEW_ARCH_ENABLED
+  [self attachReactEventListener];
```
Comment (tomekzaw) [inline packages/react-native-reanimated/apple/reanimated/apple/REAModule.mm:307]: Why do we need this change here?

Comment (tjzel) [reply packages/react-native-reanimated/apple/reanimated/apple/REAModule.mm:307]: We'd call it on Fabirc before `commonInit` here https://github.com/software-mansion/react-native-reanimated/pull/6783/files#diff-06f7028a5f8a14ac525b70098bbf48937abe91707aea813bc4528ec9c166f5bcL296 so I moved it into `commonInit` fabric section.

File: packages/react-native-reanimated/apple/reanimated/apple/REAModule.mm (hunk 2 above)
Comment (tjzel) [reply packages/react-native-reanimated/apple/reanimated/apple/REAModule.mm]: [ff3eeb2](https://github.com/software-mansion/react-native-reanimated/pull/6783/commits/ff3eeb2ec53004767ab8aa97c08813fc12fec41d)

**Review by piaskowyk — APPROVED** [review]
Now looks good to me 👍

**Review by tomekzaw — APPROVED** [review]

**Pull request merged**